pkg/api/validation/schema.go

```diff
@@ -88,6 +101,29 @@ func (s *SwaggerSchema) ValidateBytes(data []byte) error {
 	if kind == nil {
 		return fmt.Errorf("kind not set")
 	}
+	if kind.(string) == "List" {
```

Comment:
Would prefer to unmarshal to Unstructured have a place to attach other
behaviors in the future (unstructured is already able to do kind and
version via metadata).

On Sep 30, 2015, at 12:25 AM, Daniel Smith <notifications@github.com> wrote:

In pkg/api/validation/schema.go
<https://github.com/kubernetes/kubernetes/pull/14726#discussion_r40738555>:

> @@ -88,6 +101,29 @@ func (s *SwaggerSchema) ValidateBytes(data []byte) error {
>  	if kind == nil {
>  		return fmt.Errorf("kind not set")
>  	}
> +	if kind.(string) == "List" {

This should ideally work for any of the list types. Currently you can test
for this with strings.HasSuffix(kind.(string), "List"). There may be a less
brittle way.

@smarterclayton <https://github.com/smarterclayton> & @bgrant0607
<https://github.com/bgrant0607> note that this is not running on our
regular objects, it's running on the output of unmarshalling into an
interface{}-- map[string]interface{}

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/14726/files#r40738555>.